src/client/Microsoft.Identity.Client/Internal/Requests/ClientCredentialRequest.cs

```diff
@@ -86,6 +83,24 @@ protected override async Task<AuthenticationResult> ExecuteAsync(CancellationTok
             }
         }
 
+        private void validateRequest()
+        {
+            if (AuthenticationRequestParameters.Scope == null || !AuthenticationRequestParameters.Scope.Any())
+            {
+                throw new MsalClientException(
+                    MsalError.ScopesRequired,
+                    MsalErrorMessage.ScopesRequired);
+            }
+
+            // Throw exception if WithAzureRegion is set to true and the cloud is not public.
+            if (_clientParameters.WithAzureRegion && !KnownMetadataProvider.isKnownPublicEnvironment(AuthenticationRequestParameters.AuthorityInfo.Host))
```

Comment:
I don't understand this validation: 

1. we should not enforce the fact that this feature is only supported in public cloud in MSAL, it's not our job. They may support it in other clouds.
2. what would happen if the authority would not be known to MSAL (since MSAL does not have an authoritative list of known authorities)? Can't we still regionalize it?

Reply:
Discussed offline. Removed the check.